.github/workflows/gen-openapi.yml

```diff
@@ -0,0 +1,27 @@
+name: API
+
+on: [push]
+
+jobs:
+  gen-openapi:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/cargo@v1
+        with:
+          command: run
+          args: --bin gen-openapi
+      - uses: actions/upload-artifact@v3
+        with:
+          name: API schemas
+          path: api/*
+      - name: "Update API schema"
```

Comment:
I suggest to use some github action instead of writing it manually. For example: https://github.com/marketplace/actions/git-auto-commit (it has large number of stars)